### DISCUSSION TITLE
配置文件中protocol问题

### DISCUSSION THREAD

**Discussion by ClemenceCc** [baidu/dperf#412] [Q&A] [answered]
https://dperf.org/doc/html/configuration 中关于 protocol说明

TCP or UDP or HTTP protocol. Regardless of the TCP or UDP protocol, the dperf client sends an HTTP Request, and the dperf server sends an HTTP Response. Currently, 'http' only support client mode, you could use it to test nginx.

如果设置为UDP，也会发送http请求。很不明白什么意思

**Comment by pengjianzhang** [comment, marked as answer]
payload是HTTP request格式

**Comment by ClemenceCc** [comment]
另外请问，速度统计中，tx和rx，是计算的大小是仅仅payload长度，还是包含了数据包的

**Reply by pengjianzhang** [reply]
包含L2-L4头部
